src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -146,10 +147,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 });
             }
         } elseif ($options['multiple']) {
-            // <select> tag with "multiple" option
+            // "select", "text" or "hidden" widget with "multiple" option
             $builder->addViewTransformer(new ChoicesToValuesTransformer($choiceList));
+
+            // for "text" / "hidden" widget, view data uses a delimiter
+            if (in_array($options['widget'], array('text', 'hidden'))) {
+                $builder->addViewTransformer(new ValuesToStringTransformer($options['delimiter'], $options['trim']));
+            }
```

Comment:
This implementation is wrong IMO since transformers may be executed in the right order for transformation but not for reversing transformation.